src/vite/index.ts

```diff
@@ -8,9 +10,36 @@ export function getVitePlugin <UserOptions = {}> (
     const meta: UnpluginContextMeta = {
       framework: 'vite'
     }
+
     const rawPlugin = factory(userOptions, meta)
 
-    const plugin = toRollupPlugin(rawPlugin, false) as VitePlugin
+    // On windows, Vite replaces backward slashes in paths/ids with forward slashes
+    // in the `load`, `transformInclude`, and `transform` hooks. The `resolveId` hook
+    // however has backwards slashes.
+    // To ensure consistent ids across all hooks, we replace the forward slashes
+    // in the `load`, `transformInclude`, and `transform` hooks with backward slashes
+    if (os.platform() === 'win32') {
+      if (rawPlugin.load) {
+        const _load = rawPlugin.load
+        rawPlugin.load = function (id, ...args) {
+          return _load.call(this, normalizeAbsolutePath(id), ...args)
+        }
+      }
+      if (rawPlugin.transformInclude) {
+        const _transformInclude = rawPlugin.transformInclude
+        rawPlugin.transformInclude = function (id, ...args) {
+          return _transformInclude.call(this, normalizeAbsolutePath(id), ...args)
+        }
+      }
+      if (rawPlugin.transform) {
+        const _transform = rawPlugin.transform
+        rawPlugin.transform = function (code, id, ...args) {
+          return _transform.call(this, code, normalizeAbsolutePath(id), ...args)
+        }
+      }
+    }
```

Comment:
I am not sure about this part. I think in `unplugin` we should always normalize the ids to use `/`. Also, we should not alter the behavior of Vite/Rollup plugins as we are extending from them. If there is an inconsistency in Vite, we'd better to raise an issue and fix on the Vite side.

Reply:
You're totally right. Your comment made me rethink what I was actually testing. It shouldn't really matter to plugins that Vite passes different looking paths than other bundlers, as long as they're still absolute paths.

Because of this, I removed the section in question and rewrote the tests so that they're only checking whether the `transformInclude`, `transform` and `load` hooks have been called with the same ids: bd4d4696e6346e87e817c145021f08cee6a50c7f

I will follow up with an issue and reproduction example in the Vite repo so that we can track this behaviour there (and determine whether this is even something we want to change).